File: interface/src/app/core/services/project.service.ts
import { Injectable } from '@angular/core';
import {ApiService} from './api.service';
import {Observable} from 'rxjs';
import {Project} from '../../model/project';

@Injectable({
  providedIn: 'root'
})
export class ProjectService {

  constructor(private apiService: ApiService) { }
  getProjects(): Observable<Project[]> {
    return this.apiService.get('/projects');
  }
  getProjectById(projectId: number): Observable<Project> {
    return this.apiService.get('/projects/' + projectId);
  }
  saveProject(project: Project): Observable<Project> {
    return this.apiService.post('/projects', project);
  }
}

File: src/main/java/ru/ase/ims/enomanager/service/ProjectService.java
package ru.ase.ims.enomanager.service;

import ru.ase.ims.enomanager.model.Project;

import java.util.List;
import java.util.Optional;

public interface ProjectService {
    List<Project> getAllProjects();
    Optional<Project> getProjectById(Long projectId);
    Project createProject(Project project);
}
